src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
```

Comment:
Please rename variables `def_capacity` and `empty_arraydata` to be consistent with `CONSTANT_CASE`.
Ref: https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names

Reply:
Also, please avoid abbreviation of variable names like `default` instead of `def`.